src/certificate/commands/IvaCertificateCommand.ts

```diff
@@ -0,0 +1,71 @@
+import moment from 'moment';
+import { Service } from 'typedi';
+import { DefectService } from '../../defect/DefectService';
+import { IRequiredStandard, ITestResult } from '../../models';
+import { ICertificatePayload } from '../../models';
+import { CERTIFICATE_DATA, IVA_30 } from '../../models/Enums';
+import { TestResultService } from '../../test-result/TestResultService';
+import { ICertificatePayloadCommand } from '../ICertificatePayloadCommand';
+
+@Service()
+export class IvaCertificateCommand implements ICertificatePayloadCommand {
+	private type?: CERTIFICATE_DATA;
+
+	constructor(
+		private defectService: DefectService,
+		private testResultService: TestResultService
+	) {}
+
+	private certificateIsAnIva = (): boolean => this.type === CERTIFICATE_DATA.IVA_DATA;
+
+	initialise(type: CERTIFICATE_DATA, isWelsh = false) {
+		this.type = type;
+	}
+
+	public generate(testResult: ITestResult): Promise<ICertificatePayload> {
+		if (!this.certificateIsAnIva()) {
+			return Promise.resolve({} as ICertificatePayload);
```

Comment:
For consistency is it possible to make this generate method async please, removing the need for the Promise.resolve() wrapper in this return and the return on line 51.

Reply:
Ah I believe I had it this way because of the same linting reason you mentioned above.

Good spot, I'll update now 🙏